src/HistoryDialog.vala

```diff
@@ -97,9 +98,13 @@ namespace PantheonCalculator {
 
             get_content_area ().add (main_grid);
 
-            add_button (_("Close"), Gtk.ResponseType.CLOSE);
+            // Use progressive custom response code with 0 as the default close action
+            var button_clear = add_button (_("Clear History"), -1);
+            button_clear.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
 
-            var button_add = add_button (_("Insert"), Gtk.ResponseType.OK);
+            add_button (_("Close"), 0);
+
+            var button_add = add_button (_("Insert"), 1);
```

Comment:
Same here with the insert action. `Accept` or `Apply` is probably close enough that it would still be more legible than an integer

Reply:
Honestly I agree... I used all integers just for cohesion but I'll update the PR to use a custom integer only for the "Clear History" action.